### ISSUE TITLE
demod_reference_signals BLOCK (no output)

### ISSUE THREAD

**Issue by ihab9187** [BogdanDIA/gr-dvbt#19]
Hi,

I’m trying to get a signal live from an antenna on the roof through a cable connected to my SDR (hackrf one ) in ireland. The signal is 8 mhz width . So currently I’m more interesting in the receiver implantation side, i’m using dvbt_rx_demo_8k_QAM64_rate78.grc flow graph. The problem i have is that no output data coming out after the demod_reference_signals BLOCK(no cancellation map).

I have read a lot about DVB , mapping ,ofdm and others, i have spent weeks on this.

So please i need a some help with this problem . My question is why there is no output coming from the demod_reference_signals BLOCK and what is the best approach to come over this problem. any help or hints will be very helpful for me. Thanks in advance

Ihab

**Comment by nits-skydeploy** [comment]
hi there, I am also facing the same situation as yours. I am using LimeSDR as Tx and RTL SDR as Rx and not getting the constellation after demod reference signals block. Any solution, kindly share.
Regards

Nitin
www.skydeploy.com

**Comment by anthonynacz** [comment]
Hello Ihab9187 and nits-skydeploy , have you solved your problem ? I am using hackRF for tx and rx of dvbt using bogdan's flowgraphs , i do not get constellation either . I keep getting "restart acquisition" and "frequency offset" messages . When i simulate this through a file , it works perfectly , and it starts to deteriorate when i increase the noise power , and starts giving me the same "restart acquisition" and TPS lost sync messages . 
So i'm guessing it is a noise problem in the channel , but how come ? my Tx and Rx devices are barely 1 meter apart . I appreciate any help

**Comment by anthonynacz** [comment]
Hello Ihab9187 and nits-skydeploy , have you solved your problem ? I am using hackRF for tx and rx of dvbt using bogdan's flowgraphs , i do not get constellation either . I keep getting "restart acquisition" and "frequency offset" messages . When i simulate this through a file , it works perfectly , and it starts to deteriorate when i increase the noise power , and starts giving me the same "restart acquisition" and TPS lost sync messages . 
So i'm guessing it is a noise problem in the channel , but how come ? my Tx and Rx devices are barely 1 meter apart . I appreciate any help

**Comment by anthonynacz** [comment]
@ihab9187 @nits-skydeploy

**Comment by nits-skydeploy** [comment]
sorry for late reply, @anthonynacz , there is an option of Frequency correction (ppm) in rtl-sdr source block I kept it's value around 44/45 cos this is the cumulative correction for rtl. U may check that for HackRF, for RTL it is done using rtltest command. 
I consulted various people they told me that they do not use cheap dongles like RTL for receiver and used only professional Rx, so they did not face this issue. But as far as RTL is concerned this method worked for me. You may give it a shot. Best of luck, reply if u get anything.
Regards
Nitin
www.skydeploy.com

**Comment by anthonynacz** [comment]
Thanks @nits-skydeploy , yes i actually entered 20 ppm for correction for hackrf , and i am able to receive output to a file but only when i put 5MHz bandwidth and 5Msps. However it is very erroneous , because every now and then it looses TPS sync : "TPS sync NOT OK ".
 If i increase sample rate i get alot of Overruns and no output .And if i decrease sample rate and bandwidth , i start getting frequency offset . I am also experimenting with the multiply const . Have you experienced anything like these issues ?
Thank you

**Comment by nits-skydeploy** [comment]
Yeah, I m using only 1Msps, with 5Mhz. With these set of values I don't get that error u r having.
Did try the multiply constant but did not work for me. What gain do u use for Tx ad Rx. I used 30 and 50 respectively.
Regards
Nitin
www.skydeploy.com

**Comment by anthonynacz** [comment]
For Tx I'm using 20 for each of BB , IF and RF gain , the gains are separate in the osmocom block . Same for Rx . However the signal is always received with -80 dB , and doesn't get better no matter what .  I do not know if this value is two low . And regarding the multiply constant , i put a GUI slider on the Tx and tested values and found the optimal one for me . What remains to do is find a way to get rid of frequency offset error to be able to go down in sample rate .  Thank you for your insights Nitin .
